File: Others/medium/ValidStartingCity.py
def validStartingCity(distances, fuel, mpg):
    # Write your code here.

    num_cities = len(distances)
    miles = 0

    best_index = 0
    miles_at_starting = 0

    for i in range(1, num_cities):
        dst = distances[i - 1]
        fuel_prev = fuel[i - 1]

        miles += fuel_prev * mpg - dst

        if miles < miles_at_starting:
            miles_at_starting = miles
            best_index = i
    return best_index

File: Golang/medium/valid-starting-city.go
package main

func ValidStartingCity(distances []int, fuel []int, mpg int) int {
	bestNeg := 0
	bestStIdx := 0
	currSum := 0

	for idx := 0; idx < len(distances); idx++ {
		currSum += fuel[idx]*mpg - distances[idx]
		if currSum < bestNeg {
			bestNeg = currSum
			bestStIdx = (idx + 1) % len(distances)
		}
	}
	return bestStIdx
}
